File: _plugins/amazon_url.rb
# Amazon URL Tag
# usage: {% amazon_url http://amazon.com/ %}

module Jekyll
  class AmazonUrlTag < Liquid::Tag

    def initialize(tag_name, url, tokens)
      super
      @url = url.strip
    end

    def render(context)
      # amazon_tag = context.environments.first["site"]["amazon_tag"]
      amazon_tag = context.registers[:site].config["amazon_tag"]

      "#{@url}?tag=#{amazon_tag}"
    end
  end
end

Liquid::Template.register_tag('amazon_url', Jekyll::AmazonUrlTag)

File: themes/jonsuh-v8/scripts/amazon_url.js
/*
  {% amazon_url ITEM_ID %}
  {% amazon_url https://www.amazon.com/dp/ITEM_ID %}
*/
const url = require('url');

hexo.extend.tag.register('amazon_url', function(args, content) {
  let amazonUrl = args[0];

  if ( ! amazonUrl.includes("://")) {
    amazonUrl = url.resolve("https://www.amazon.com/dp/", amazonUrl);
  }

  const tagQuery = "?tag=" + hexo.config.amazon_tag;

  return url.resolve(amazonUrl, tagQuery);
}, {ends: false});
